src/SimplePDPService.sol

```diff
@@ -190,24 +195,38 @@ contract SimplePDPService is PDPListener, PDPRecordKeeper, Initializable, UUPSUp
     // nextProvingPeriod checks for unsubmitted proof and emits a fault if so
     function nextProvingPeriod(uint256 proofSetId, uint256 challengeEpoch, uint256 leafCount, bytes calldata) external onlyPDPVerifier {
         receiveProofSetEvent(proofSetId, OperationType.NEXT_PROVING_PERIOD, abi.encode(challengeEpoch, leafCount));
-        // Noop when proving period not yet open
+
+        // initialize state for new proofset
+        if (provingDeadlines[proofSetId] == NO_PROVING_DEADLINE) {
```

Comment:
Can you explain what we gain from this?  Since we're checking that the epoch is within the challenge window that will already cover this check right?

Reply:
okay yeah fair enough. Didnt realise this.